insights/parsers/journalctl.py

```diff
@@ -0,0 +1,154 @@
+"""
+JournalCtl - command ``journalctl xxx``
+=======================================
+
+This module contains the following parsers:
+
+JournalAll - file ``/sos_commands/logs/journalctl_--no-pager``
+--------------------------------------------------------------
+JournalctlHeader - command ``/usr/bin/journalctl --header``
+-----------------------------------------------------------
+JournalSinceBoot - file ``/sos_commands/logs/journalctl_--no-pager_--boot``
+---------------------------------------------------------------------------
+
+"""
+
+
+from insights.core.plugins import parser
+from insights.core import CommandParser, LegacyItemAccess
+from insights.parsers import SkipException, get_active_lines
+from insights.specs import Specs
+from .. import Syslog
+
+
+@parser(Specs.journal_all)
+class JournalAll(Syslog):
+    """
+    Read the ``/sos_commands/logs/journalctl_--no-pager`` file.  Uses the
+    ``Syslog`` class parser functionality - see the base class for more details.
+
+    Sample log lines::
+
+        -- Logs begin at Wed 2017-02-08 15:18:00 CET, end at Tue 2017-09-19 09:25:27 CEST. --
+        May 18 15:13:34 lxc-rhel68-sat56 jabberd/sm[11057]: session started: jid=rhn-dispatcher-sat@lxc-rhel6-sat56.redhat.com/superclient
+        May 18 15:13:36 lxc-rhel68-sat56 wrapper[11375]: --> Wrapper Started as Daemon
+        May 18 15:13:36 lxc-rhel68-sat56 wrapper[11375]: Launching a JVM...
+        May 18 15:24:28 lxc-rhel68-sat56 yum[11597]: Installed: lynx-2.8.6-27.el6.x86_64
+        May 18 15:36:19 lxc-rhel68-sat56 yum[11954]: Updated: sos-3.2-40.el6.noarch
+
+    .. note::
+        Because journal timestamps by default have no year,
+        the year of the logs will be inferred from the year in your timestamp.
+        This will also work around December/January crossovers.
+
+    Examples:
+        >>> type(JournalAll)
+        <class 'insights.parsers.journalctl.JournalAll'>
+        >>> len(JournalAll.lines)
+        10
+        >>> bona_list = JournalAll.get('(root) LIST (root)')
+        >>> bona_list[0].get('message')
+        '(root) LIST (root)'
+    """
+    pass
+
+
+@parser(Specs.journalctl_header)
+class JournalctlHeader(LegacyItemAccess, CommandParser):
```

Comment:
@TZ3070 - The `LegacyItemAccess` is not recommended anymore, if it's necessary , you can directly mix the parser with `list` or `dict` accordingly.

However, here I don't think it's required to parse the `journalctl_header` to a dict, since no rules read the accurate lines. You can just inherit it from the `Syslog` just like the other two.